### PR TITLE
Clean up ARM64

### PR DESCRIPTION
**Pull request by klauspost**
Update private package.
Remove extensive flag names.
Use separate type for ARM flags.
Simpler/more extensible interface for populating CPU info.
Don't specify ARM 64, since 32 bit support may be added later.

### CODE REVIEW

**Comment by klauspost** [comment]
PTAL @fwessels

File: cpuid.go [hunk 1]
```diff
@@ -751,18 +747,35 @@ func (f Flags) String() string {
 
 // Strings returns and array of the detected features.
```
Comment (fwessels) [inline cpuid.go:748]: returns and array --> returns an array

File: private/cpuid.go [hunk 2]
```diff
@@ -687,18 +741,35 @@ func (f flags) String() string {
 
 // Strings returns and array of the detected features.
```
Comment (fwessels) [inline private/cpuid.go:742]: and --> an

File: private/cpuid.go [hunk 3]
```diff
@@ -687,18 +741,35 @@ func (f flags) String() string {
...
 	return r
 }
 
+// String returns a string representation of the detected
+// CPU features.
+func (f armflags) String() string {
+	return strings.Join(f.strings(), ",")
+}
+
+// Strings returns and array of the detected features.
```
Comment (fwessels) [inline private/cpuid.go:761]: idem

**Pull request merged**